qiskit/opflow/converters/circuit_sampler.py

```diff
@@ -251,21 +246,26 @@ def clear_cache(self) -> None:
         """Clear the cache of sampled operator expressions."""
         self._cached_ops = {}
 
-    def _extract_circuitstatefns(self, operator: OperatorBase) -> None:
+    def _extract_circuitstatefns(
+        self, operator: OperatorBase, result: Dict[int, CircuitStateFn]
+    ) -> None:
         r"""
         Recursively extract the ``CircuitStateFns`` contained in operator into the
         ``_circuit_ops_cache`` field.
         """
         if isinstance(operator, CircuitStateFn):
-            self._circuit_ops_cache[id(operator)] = operator
+            result[id(operator)] = operator
         elif isinstance(operator, ListOp):
             for op in operator.oplist:
-                self._extract_circuitstatefns(op)
+                self._extract_circuitstatefns(op, result)
 
+    # pylint: disable=unused-argument
+    @deprecate_arguments({"circuit_sfns": "operator"})
     def sample_circuits(
```

Comment:
I am wondering about the extent of the changes here. Should we be keeping `sample_circuits` for people to use as it was and add a new method that takes operator to do what we want now. If I had done code that used sample_circuits - its a public interface here after all - I think the changes here means it no longer works for my existing code - deprecated means it should keep on working as it had done until we remove the deprecated function.

Reply:
It is true. However, I personally feel uncomfortable that this method, which has no unit tests, is public. Considering the future maintenance, I think the method should be deprecated now. I can't think of a use case for this method by itself.

Reply:
@woodsp-ibm Thanks a lot. I finally decided not to change this method.